scripts/steps/build_and_test_with_xcodebuild.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -x
+
+# Builds, then tests the project using the specified destination
+# using the Xcode command line tools (xcodebuild).
+
+# Remove any Swift/SPM build artifacts
+# (these have been causing conflicts in recent Xcode versions)
+rm -rf .build
+
+# Perform the build using the xcodebuild CLI tool.
+# Create a temp dir for derived data (this seems to help
+# with certain spurious Xcode failures.)
+# Pipe the build logs through xcpretty.
+# On build failure, fail on Xcode's status code
+xcodebuild \
+  -scheme AWSSwiftSDK-Package \
+  -destination "$AWS_SDK_XCODEBUILD_DESTINATION" \
+  -derivedDataPath `mktemp -d` \
+  build test \
+  | xcpretty && exit ${PIPESTATUS[0]}
```

Comment:
how is it made sure that you have xcpretty installed?

Reply:
It is installed on Github CI runners by default.  To run this on desktop, we can either require that developers install it (it is distributed as a Rubygem), or I can test for `xcpretty` and not pipe through it if it's not installed.

Xcode build logs are pretty unwieldy without it though (Github actually truncates our build logs if you don't use it), so probably better to require it to be installed.